Sources/VitalCore/Core/Client/Data Models/Base Models/VitalResource.swift

```diff
@@ -136,3 +169,33 @@ public enum VitalResource: Equatable, Hashable, Codable {
     }
   }
 }
+
+public enum BackfillType: String, Codable {
```

Comment:
```suggestion
@_spi(VitalSDKInternals)
public enum BackfillType: String, Codable {
```

Reply:
I would define this as a struct:

```
public struct BackfillType: String, RawRepresentable, Codable {
  public let rawValue: String

  public init(rawValue: String) {
    self.rawValue = rawValue
  }

  public static let workouts = BackfillType(rawValue: "workouts")
  public static let body = BackfillType(rawValue: "body")
  // ...
}
```

Enum is exhaustive so if the sync state endpoint starts returning new BackfillTypes, decoding is going to fail.
(unless we make the sync state endpoint conditionalize on the SDK version header...)

Reply:
so in the case of the struct, in `resourceToBackfillType` mapping we would do:
```swift
default:
      return BackfillType.rawValue;
```
?

Reply:
That shouldn’t need a default case. VitalResource is also exhaustive but the SDK is always built from source, so you won’t have a new case going into an old codepath.